content/blog/espressif-ide-lsp-support/index.md

```diff
@@ -0,0 +1,227 @@
+---
+title: "Espressif-IDE 3.0.0 — LSP Support for C/C++ Editor"
+date: 2024-06-28
+showAuthor: false
+tags: ["News", "Espressif IDE", "LLVM", "Eclipse"]
+authors:
+  - "kondal-kolipaka"
+---
+
+This article has the following major sections:
+
+- [Introduction](#introduction)
+- [Deciphering the alphabet soup](#deciphering-the-alphabet-soup)
+- [Project building and toolchains](#project-building-and-toolchains)
+- [Why moving to LSP-based editor?](#why-moving-to-lsp-based-editor)
+- [Clangd Setup](#clangd-setup)
+- [Editor Features](#editor-features)
+- [Conclusion](#conclusion)
+
+
+## Introduction
+
+We are excited to announce the release of Espressif-IDE 3.0.0, a significant update that brings long-awaited features!
+
+This version includes:
+
+- **Eclipse CDT-LSP plugins** that bringing support for the latest C/C++ standards
+- New **LSP-based C/C++ Editor** powered by the LLVM clangd C/C++ language server and offering advanced editor features for ESP-IDF developers
+
+
+## Deciphering the alphabet soup
+
+The article includes a number of concepts and abbreviations that might look somewhat intimidating, such as LLVM, LSP, CDT-LSP, Clang, Clangd, esp-clang, etc. The uninitiated reader might easily get lost. Let's go through those quickly.
+
+```mermaid
+flowchart LR
+subgraph ide[Espressif-IDE]
+  ide1["ESP-IDF Eclipse Plugin\n(IEP plugin)"]
+  style ide1 fill:#99f
+  subgraph ide10[Eclipse CDT]
+    ide11[C/C++\nEditor]
+  end
+  style ide10 fill:#99f
+  ide11[C/C++\neditor] <--> ide21[LSP]
+  subgraph ide20[Eclipse CDT-LSP]
+    direction LR
+    ide21[LSP] <--> ide22[clangd]
+  end
+  style ide20 fill:#99f
+end
+subgraph idf[ESP-IDF]
+  idf2[Other\ncomponents]
+  idf1[esp-clang]
+end
+ide22[clangd\nconfig] <--> idf1[esp-clang]
+```
```

Comment:
This is a mermaid diagram that looks like this:

![image](https://github.com/espressif/developer-portal/assets/105578077/c894eba3-8e4c-4841-8a48-c73c53344487)